docroot/sites/now.uiowa.edu/modules/now_migrate/src/Plugin/migrate/source/InTheNews.php

```diff
@@ -0,0 +1,185 @@
+<?php
+
+namespace Drupal\now_migrate\Plugin\migrate\source;
+
+use Drupal\migrate\Row;
+use Drupal\sitenow_migrate\Plugin\migrate\source\BaseNodeSource;
+use Drupal\sitenow_migrate\Plugin\migrate\source\ProcessMediaTrait;
+use Drupal\taxonomy\Entity\Term;
+
+/**
+ * Migrate Source plugin.
+ *
+ * @MigrateSource(
+ *   id = "now_in_the_news",
+ *   source_module = "node"
+ * )
+ */
+class InTheNews extends BaseNodeSource {
+  use ProcessMediaTrait;
+
+  /**
+   * {@inheritdoc}
+   */
+  public function prepareRow(Row $row) {
+    parent::prepareRow($row);
+
+    // Process the primary media field.
+    $media = $row->getSourceProperty('field_primary_media');
+    if (!empty($media)) {
+      // Check if it's a video or image.
+      $filemime = $this->select('file_managed', 'fm')
+        ->fields('fm', ['filemime'])
+        ->condition('fm.fid', $media[0]['fid'], '=')
+        ->execute()
+        ->fetchField();
+      // If it's an image, we can handle it like normal.
+      if (str_starts_with($filemime, 'image')) {
+        $fid = $this->processImageField($media[0]['fid'], $media[0]['alt'], $media[0]['title']);
+        $row->setSourceProperty('field_primary_media', $fid);
+      }
+      elseif (in_array($filemime, ['video/oembed', 'application/octet-stream'])) {
+        $body = $row->getSourceProperty('body');
+        $body[0]['value'] = $this->createVideo($media[0]['fid']) . $body[0]['value'];
+        $row->setSourceProperty('body', $body);
+      }
+    }
+
+    $source_collection = $row->getSourceProperty('field_sources');
+    if (!empty($source_collection)) {
+      // Even if there are multiple sources, we can grab the first
+      // and ignore the rest.
+      $source_id = $source_collection[0]['revision_id'];
+      $url = $this->select('field_data_field_url', 'url')
+        ->fields('url', ['field_url_url'])
+        ->condition('url.revision_id', $source_id, '=')
+        ->execute()
+        ->fetchField();
+      $url = $this->fixUrls($url);
+      // @todo This could be combined into a single call.
```

Comment:
Is this going to be addressed as part of this PR or should a formal todo task be created and referenced here?

Reply:
Totally forgot about that!

Updated it to use a single query.